shanoir-ng-datasets/src/main/java/org/shanoir/ng/exporter/controller/BidsApiController.java

```diff
@@ -49,39 +52,52 @@
 	}
 
 	@Override
-    public ResponseEntity<ByteArrayResource> exportBIDSFile(
+    public void exportBIDSFile(
     		@ApiParam(value = "Id of the study", required=true) @PathVariable("studyId") Long studyId,
-    		@ApiParam(value = "file path") @Valid @RequestParam(value = "filePath", required = true) String filePath) throws RestServiceException, IOException {
+    		@ApiParam(value = "file path") @Valid @RequestParam(value = "filePath", required = true) String filePath, HttpServletResponse response) throws RestServiceException, IOException {
 		// Check filePath too
 		// /var/datasets-data/bids-data/stud-1_NATIVE
 		if (!filePath.startsWith("/var/datasets-data/bids-data/stud-" + studyId)) {
-			return new ResponseEntity<>(HttpStatus.UNAUTHORIZED);
+			response.sendError(HttpStatus.UNAUTHORIZED.value());
+			return;
 		}
 
 		// Get file, zip it and download it
     	File fileToBeZipped = new File(filePath);
     	if (!fileToBeZipped.exists()) {
-			return new ResponseEntity<>(HttpStatus.NO_CONTENT);
+			response.sendError(HttpStatus.NO_CONTENT.value());
+			return;
     	}
     	
     	// Copy / zip it (and by the way filter only folder that we are interested in)
+    	String userDir = getUserDir(System.getProperty(JAVA_IO_TMPDIR)).getAbsolutePath();
```

Comment:
Hi Jean-Come, it could be a little bit risky to use IO_TMPDIR here, as this resolves depending on the OS. As we are always in the same container using docker-compose this is not an issue, but it could happen, that his variable does not point to the /tmp, what is the named volume proposed by docker-compose.